BC.Infrastructure.Business/BusinessServices/ProjectService.cs

```diff
@@ -0,0 +1,64 @@
+﻿using BC.Domain.Core;
+using BC.Infrastructure.Data.Repository;
+using BC.Services.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace BC.Infrastructure.Business.BusinessServices
+{
+    public class ProjectService : BaseService, IService<Project>
+    {
+        private readonly Uow _uow;
+
+        public ProjectService()
+        {
+            this._uow = BaseService.GetUow();
+        }
+
+        public IEnumerable<Project> GetAll()
+        {
+            return _uow.Project.All.AsEnumerable();
+        }
+
+        public Project GetById(Guid id)
+        {
+            return _uow.Project.Find(id);
+        }
+
+        public void Add(Project project)
+        {
+            project.Id = CheckDefaultId(project);
+
+            if (project.Name == string.Empty)
+            {
+                throw new ArgumentException("Project name cant be empty");
+            }
+            _uow.Project.InsertOrUpdate(project);
+            _uow.Save();
+        }
+
+        public void Update(Project project)
+        {
+            project.Id = CheckDefaultId(project);
```

Comment:
unnecessary method here. User can't set default guid while update project. if id will change we will have  referential integrity (payments)